force-app/main/default/classes/ContactAccountHandlerTest.cls

```diff
@@ -0,0 +1,46 @@
+@isTest
+public with sharing class ContactAccountHandlerTest {
+    @isTest static void isPrimaryTest(){
+        Account acc1 = new Account(Name='Demo');
+        insert acc1;
+        
+        Account acc2 = new Account(Name='Demo 1', ParentId=acc1.Id);
+        insert acc2;
+        
+        List<Contact> contactList = new List<Contact>();
+        for(Integer i=0; i<3; i++){
+            Contact contact = new Contact();
+            contact.LastName='Demo Contact '+i;
+            contact.AccountId=acc1.id;
```

Comment:
here you are making `acc1` parent for all contacts, 
and requirement is like we account parent account email should be change to primary contact's email right?

Reply:
Yess , the contact.Email is set to ParentAccount.PrimaryContact.Email